docs/quickstarts/tanstack-start.mdx

```diff
@@ -0,0 +1,243 @@
+
+    }
+  ]}
+  beforeYouStart={[
+    {
+      title: "Set up a Clerk application",
+      link: "/docs/quickstarts/setup-clerk",
+      icon: "clerk",
+    }
+  ]}
+>
+  - Install `@clerk/tanstack-start`
+  - Set up your environment variables
+  - Configure `clerkHandler`
+  - Add `<ClerkProvider />` to your application
+  - Protect your pages
+</TutorialHero>
+
+Learn how to use Clerk to quickly and easily add secure authentication and user management to your TanStack Start application.
+
+<Steps>
+  ### Install `@clerk/tanstack-start`
+
+  Once you have a TanStack Start application ready, you need to install Clerk's TanStack Start SDK. This gives you access to our prebuilt components and hooks for TanStack Start applications.
+
+  <CodeBlockTabs type="installer" options={["npm", "yarn", "pnpm"]}>
+    ```bash {{ filename: 'terminal' }}
+    npm install @clerk/tanstack-start
+    ```
+
+    ```bash {{ filename: 'terminal' }}
+    yarn add @clerk/tanstack-start
+    ```
+
+    ```bash {{ filename: 'terminal' }}
+    pnpm add @clerk/tanstack-start
+    ```
+  </CodeBlockTabs>
+
+  ### Set your environment variables
+
+  Add the following keys to your `.env` file. These keys can always be retrieved from the [API Keys](https://dashboard.clerk.com/last-active?path=api-keys) page of your Clerk Dashboard.
+
+  ```env {{ filename: '.env' }}
+  VITE_CLERK_PUBLISHABLE_KEY={{pub_key}}
+  CLERK_SECRET_KEY={{secret}}
+  ```
+
+  ### Add `clerkHandler` to your application
+
+  To configure Clerk in your TanStack application, you will need to update the SSR entrypoint.
+
+  ```tsx {{ filename: 'app/ssr.tsx', ins: [4, [11, 14]], del: [[6, 9]] }}
+  import { createStartHandler, defaultStreamHandler } from '@tanstack/start/server'
+  import { getRouterManifest } from '@tanstack/start/router-manifest'
+  import { createRouter } from './router'
+  import { createClerkHandler } from '@clerk/tanstack-start/server'
+
+  export default createStartHandler({
+    createRouter,
+    getRouterManifest,
+  })(defaultStreamHandler)
+
+  export default createClerkHandler(
+    createStartHandler({
+      createRouter,
+      getRouterManifest,
+    }),
+  )(defaultStreamHandler)
+  ```
+
+  ### Add `<ClerkProvider>` and components to your app
+
+  All Clerk hooks and components must be children of the [`<ClerkProvider>`](/docs/components/clerk-provider) component, which provides active session and user context.
+
+  ```tsx {{ filename: 'app/routes/__root.tsx', ins: [11, 28, 32] }}
+  import { Outlet, ScrollRestoration, createRootRoute } from '@tanstack/react-router'
+  import { TanStackRouterDevtools } from '@tanstack/router-devtools'
+  import { Body, Head, Html, Meta, Scripts } from '@tanstack/start'
+  import * as React from 'react'
+  import { DefaultCatchBoundary } from '~/components/DefaultCatchBoundary'
+  import { NotFound } from '~/components/NotFound'
+  import { ClerkProvider } from '@clerk/tanstack-start'
+  import appCss from '~/styles/app.css?url'
```

Comment:
There are a lot of imports here, and things going on in this code block that are not from the tanstack quickstart - if we are going to add things like this, we need to explain them in the doc. But these don't seem necessary for a "quickstart". Can you provide more info here on what's being added and why?

Reply:
Yeah we can remove some of them, most of those are taken from the early example of TanStack Start as there was no docs when we started creating the SDK